src/pal/src/exception/signal.cpp

```diff
@@ -714,4 +783,111 @@ void restore_signal(int signal_id, struct sigaction *previousAction)
     }
 }
 
+static
+DWORD
+PALAPI
+ExternalSignalHandlerThreadRoutine(
+    PVOID
+    );
+
+PAL_ERROR
+StartExternalSignalHandlerThread(
+    CPalThread *pthr)
+{
+    PAL_ERROR palError = NO_ERROR;
+    
+#ifndef DO_NOT_USE_SIGNAL_HANDLING_THREAD
+    HANDLE hThread;
+
+    if (pipe(g_signalPipe) != 0)
+    {
+        palError = ERROR_CANNOT_MAKE;
+        goto done;
+    }
+
+    palError = InternalCreateThread(
+        pthr,
+        NULL,
+        0,
+        ExternalSignalHandlerThreadRoutine,
+        NULL,
+        0,
+        SignalHandlerThread, // Want no_suspend variant
+        &g_dwExternalSignalHandlerThreadId,
+        &hThread
+        );
+
+    if (palError != NO_ERROR)
+    {
+        ERROR("Failure creating external signal handler thread (%d)\n", palError);
+        goto done;
+    }
+
+    InternalCloseHandle(pthr, hThread);
+
+    handle_signal(SIGTERM, sigterm_handler, &g_previous_sigterm);
+#endif // DO_NOT_USE_SIGNAL_HANDLING_THREAD
+
+done:
+
+    return palError;
+}
+
+static
+DWORD
+PALAPI
+ExternalSignalHandlerThreadRoutine(
+    PVOID
+    )
+{
+    DWORD dwThreadId;
+    bool fContinue = TRUE;
+    HANDLE hThread;
+    PAL_ERROR palError = NO_ERROR;
+    CPalThread *pthr = InternalGetCurrentThread();
+
+    //
+    // Wait for a signal to occur
+    //
+
+    while (fContinue)
+    {
+        BYTE signalCode;
+        ssize_t bytesRead;
+
+        do
+        {
+            bytesRead = read(g_signalPipe[0], &signalCode, 1);
+        }
+        while ((bytesRead == -1) && (errno == EINTR));
+
+        if (bytesRead == -1)
+        {
+            // Fatal error 
+            abort();
```

Comment:
I think we might want to use PROCAbort instead of abort to shut down the debugger transport too.